### ISSUE TITLE
Bug: withQueryParams merging object instead of replacing

### ISSUE THREAD

**Issue by tiarebalbi** [supercharge/sttp#6]
Doesn't matter how you calling the `withQueryParams`  the this.request.params gets merged with previous value.

In the scenario where I had a filter and now I'm removing the filter the 
object params was:
```js
this.request.params = {
    status: 1
}
```
Now I want to remove, I will the call the `withQueryParams({})`.... the result of this operation will result in the same filter I had before.

I would suggest refactoring the following:
https://github.com/supercharge/sttp/blob/main/src/sttp.ts#L122-L133

to: 
```js
withQueryParams (queryParams: object): this {
  this.request.params = queryParams

  return this
}
```

@marcuspoehls thoughts?

My workaround was to manually override the setting

```js
_.set(instance.requestConfig(), 'params', undefined);
```

**Comment by marcuspoehls** [comment]
@tiarebalbi Hey Tiarê, thank you for reaching out. I see the problem in your use case of filters that get added and removed.

I would like to keep the `withQueryParams` method that merges new parameters with existing ones because someone may create a base client using `Sttp.create()` with a default query parameter, like an API key.

Instead, I would like to see a new method `removeQueryParams()` deleting all previously set query parameters. Then, someone could chain `sttp.removeQueryParams().withQueryParams({ my: 'filter' })`

@tiarebalbi what do you think? Are you up to creating a pull request for this?

**Comment by tiarebalbi** [comment]
That looks like a good idea, I was waiting your feedback. Let me get the PR ready for this change. Thanks @marcuspoehls

**Comment by marcuspoehls** [comment]
@tiarebalbi I’ve published version `1.5.0` containing the added `removeQueryParams` method 🙂  thank you again for contributing this!
